src/TRestGeant4Metadata.cxx

```diff
@@ -1418,19 +1459,19 @@ Int_t TRestGeant4Metadata::ReadOldDecay0File(TString fileName) {
 void TRestGeant4Metadata::ReadParticleSource(TiXmlElement* definition) {
     TiXmlElement* sourceDefinition = definition;
 
-    TRestGeant4ParticleSource source;
+    TRestGeant4ParticleSource* source = new TRestGeant4ParticleSource();
```

Comment:
The `new` could lead to memory leak, I cannot see source being deleted...

Reply:
Actually the method is never called. I have a new method `ReadParticleSource(TRestGeant4ParticleSource* source, TiXmlElement* definition)` to initialize particle source.

For the memory leaking, I think we can just delete particle sources at the destructor of TRestGeant4Metadata